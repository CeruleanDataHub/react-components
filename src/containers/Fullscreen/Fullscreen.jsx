import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { Icon } from "../Icon/Icon";

const Container = styled.span`
  cursor: pointer;
`;

export const Fullscreen = ({ as, onClick }) => (
  <Container
    onClick={onClick}
    onKeyPress={onClick}
    as={as}
    role="button"
    tabIndex={0}
  >
    <Icon name="full-screen" />
  </Container>
);

Fullscreen.propTypes = {
  /** Extend styles, property is from styled components */
  as: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.string,
    PropTypes.shape({ render: PropTypes.func.isRequired })
  ]),
  /** Click event listener */
  onClick: PropTypes.func
};

Fullscreen.defaultProps = {
  as: null,
  onClick: () => null
};
