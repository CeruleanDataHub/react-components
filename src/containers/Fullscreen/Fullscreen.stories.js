import { action } from "@storybook/addon-actions";
import React from "react";

import { Fullscreen } from "./Fullscreen";

export default {
  title: "Fullscreen",
  component: Fullscreen
};

export const ChartStory = () => (
  <Fullscreen onClick={action("button clicked")} />
);
