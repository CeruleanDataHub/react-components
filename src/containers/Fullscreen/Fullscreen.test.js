import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import { Fullscreen } from "./Fullscreen";

describe("Fullscreen", () => {
  it("should render", () => {
    const component = renderer.create(<Fullscreen />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should have onClick listener", () => {
    const component = renderer.create(<Fullscreen onClick={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should fire onClick event callback function", () => {
    const handleClick = jest.fn();
    const component = shallow(<Fullscreen onClick={handleClick} />);
    expect(handleClick).not.toHaveBeenCalled();
    component.simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });

  it("should call default props onClick function when no onClick property is passed", () => {
    const component = renderer.create(<Fullscreen />);
    component.root.props.onClick();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
