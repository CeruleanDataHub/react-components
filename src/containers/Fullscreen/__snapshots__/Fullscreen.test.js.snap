// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fullscreen should call default props onClick function when no onClick property is passed 1`] = `
.c1 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  cursor: pointer;
}

<span
  className="c0"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="c1 lni lni-full-screen"
  />
</span>
`;

exports[`Fullscreen should have onClick listener 1`] = `
.c1 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  cursor: pointer;
}

<span
  className="c0"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="c1 lni lni-full-screen"
  />
</span>
`;

exports[`Fullscreen should render 1`] = `
.c1 {
  font-size: 4rem;
  text-align: center;
}

.c0 {
  cursor: pointer;
}

<span
  className="c0"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="c1 lni lni-full-screen"
  />
</span>
`;
